substratest/factory.py

```diff
@@ -371,13 +383,14 @@ def add_aggregatetuple(self, aggregate_algo, worker, in_models=None, tag=''):
             worker=worker,
             in_models_ids=[t.id for t in in_models],
             tag=tag,
+            metadata=metadata or {}
```

Comment:
In half the cases, you've been doing `metadata=metadata` and in the other half `metadata=metadata or {}`. You should chose one of the two solutions and stick with it. When you do, could you also explain why you chose one over the other?